### PR TITLE
chore: Uses zero values for computed attributes and avoids `""` in payloads

### PR DESCRIPTION
**Pull request by EspenAlbert**
## Description

- Uses zero values for computed attributes and avoids `""` in payloads

Link to any related issue(s): CLOUDP-287995

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/common/conversion/type_conversion.go [hunk 1]
```diff
@@ -3,8 +3,18 @@ package conversion
...
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
 )
 
+func SafeValue[T any](v *T) T {
+	if v != nil {
+		return *v
+	}
+	emptyValue := new(T)
+	return *emptyValue
```
Comment (lantoli) [inline internal/common/conversion/type_conversion.go]:
```suggestion
	var emptyValue T
	return emptyValue
```

that is good, curious if this alternative also works

File: internal/common/conversion/type_conversion.go [hunk 2]
```diff
@@ -3,8 +3,18 @@ package conversion
...
+}
+
 func SafeString(s *string) string {
```
Comment (lantoli) [inline internal/common/conversion/type_conversion.go:18]: we can remove SafeString and use SafeValue instead

File: internal/common/conversion/type_conversion.go [hunk 3]
```diff
@@ -86,3 +96,11 @@ func NilForUnknown[T any](primitiveAttr TFPrimitiveType, value *T) *T {
 	}
 	return value
 }
+
+func NilForUnknownOrEmpty(primitiveAttr types.String) *string {
```
Comment (lantoli) [inline internal/common/conversion/type_conversion.go]: should we add in the func name "string" as we might want similar funcs for ints, floats, or bools?

File: internal/common/conversion/type_conversion_test.go [hunk 4]
```diff
@@ -95,3 +96,22 @@ func TestAWSRegionToMongoDBRegion(t *testing.T) {
...
+
+func TestSafeValue(t *testing.T) {
+	var boolPointer *bool
+	assert.False(t, conversion.SafeValue(boolPointer))
+	trueBool := true
+	assert.True(t, conversion.SafeValue(&trueBool))
+	var intPointer *int
+	assert.Equal(t, 0, conversion.SafeValue(intPointer))
+	var stringPointer *string
+	assert.Equal(t, "", conversion.SafeValue(stringPointer))
```
Comment (lantoli) [inline internal/common/conversion/type_conversion_test.go:108]: good test showing that we don't need SafeString :-)

File: internal/service/advancedclustertpf/model_ClusterDescription20240805.go [hunk 5]
```diff
@@ -38,32 +38,32 @@ func NewTFModel(ctx context.Context, input *admin.ClusterDescription20240805, ti
 	}
 	return &TFModel{
 		AcceptDataRisksAndForceReplicaSetReconfig: types.StringPointerValue(conversion.TimePtrToStringPtr(input.AcceptDataRisksAndForceReplicaSetReconfig)),
-		BackupEnabled:                    types.BoolPointerValue(input.BackupEnabled),
+		BackupEnabled:                    types.BoolValue(conversion.SafeValue(input.BackupEnabled)),
 		BiConnectorConfig:                biConnector,
-		ClusterType:                      types.StringPointerValue(input.ClusterType),
-		ConfigServerManagementMode:       types.StringPointerValue(input.ConfigServerManagementMode),
-		ConfigServerType:                 types.StringPointerValue(input.ConfigServerType),
+		ClusterType:                      types.StringValue(conversion.SafeValue(input.ClusterType)),
```
Comment (lantoli) [inline internal/service/advancedclustertpf/model_ClusterDescription20240805.go:43]: nice

**Review by lantoli — APPROVED** [review]
LGTM

File: internal/common/conversion/type_conversion.go (hunk 1 above)
Comment (EspenAlbert) [reply internal/common/conversion/type_conversion.go]: Good idea. Changed in 4b16f455ea32f9f74f93c78d8fcbc9c889589139

File: internal/common/conversion/type_conversion.go (hunk 2 above)
Comment (EspenAlbert) [reply internal/common/conversion/type_conversion.go:18]: It is used in 11 files... Prefer doing it in a separate PR

File: internal/common/conversion/type_conversion.go (hunk 3 above)
Comment (EspenAlbert) [reply internal/common/conversion/type_conversion.go]: I like it, changed in 4b16f455ea32f9f74f93c78d8fcbc9c889589139

**Pull request merged**